projects/sandbox/train/train/train.py

```diff
@@ -212,43 +209,25 @@ def main(
 
     # load background, infer ifos, and get start and end times
     # of the combined training + validation period
-    logging.info("Loading background data")
-    background, ifos, t0 = train_utils.get_background(background_dir)
-    duration = background.shape[-1] / sample_rate
-    logging.info(
-        "Loaded {} seconds of data from ifos {}".format(
-            duration, ", ".join(ifos)
-        )
+    background_fnames = train_utils.get_background_fnames(background_dir)
+
+    sample_length = kernel_length + psd_length
+
+    psd_estimator = structures.PsdEstimator(
+        psd_length, sample_rate, fftlength=2, fast=highpass is not None
     )
+    whitener = preprocessor.Whitener(fduration, sample_rate)
+    whitener = whitener.to(device)
 
     # load our waveforms and build some objects
     # for augmenting their snrs
     waveforms, valid_waveforms = train_utils.get_waveforms(
         waveform_dataset, ifos, sample_rate, valid_frac
     )
-    waveform_duration = waveforms.shape[-1] / sample_rate
-    rescaler = SnrRescaler(len(ifos), sample_rate, waveform_duration, highpass)
-    snr_sampler = SnrSampler(
-        max_min_snr, snr_thresh, max_snr, snr_alpha, snr_decay_steps
-    )
-
-    if valid_frac is not None:
-        background, valid_background = train_utils.split(
-            background, 1 - valid_frac, -1
-        )
-        tf = t0 + (1 - valid_frac) * duration
-
-        # fit our rescaler here since we'll need its
-        # background psd estimate to threshold
-        logging.info("Rescaling validation waveforms")
-        rescaler.fit(*background)
-        valid_waveforms = train_utils.threshold_snrs(
-            valid_waveforms,
-            snr_thresh,
-            sample_rate,
-            rescaler.background,
-            highpass,
-        )
+    if valid_waveforms is not None:
+        valid_fname = background_fnames.pop(-1)
```

Comment:
Did we switch to having the validation dataset just be the last segment from the training time?

Reply:
Yeah exactly, but we do timeslides so it doesn't affect the amount of background we generate